scripts/saveReleaseNotes.plugin.js

```diff
@@ -0,0 +1,170 @@
+import fs from 'fs';
+
+/**
+ * @param {Object} config
+ * @param {import('semantic-release').GenerateNotesContext} context
+ * */
+export async function generateNotes(config, context) {
+  const parsedNotes = parseNotes(context.nextRelease.notes, context.commits);
+
+  fs.writeFileSync('./release-notes.json', JSON.stringify(parsedNotes));
+}
+
+const HEADER_INDICATOR = '###';
+const BREAKING_CHANGES_HEADER = 'breaking-changes';
+
+class InvalidReleaseNoteError extends Error {
+  constructor() {
+    super('Release note format is invalid or does not match expected pattern.');
+  }
+}
+
+/**
+ * @param {string} notes
+ * @param {import('semantic-release').Commit[]} commits
+ * */
+export function parseNotes(notes, commits) {
+  let currentHeader = '';
+  const result = {};
+
+  const seenNotes = new Set();
+  const notesShaMap = new Map();
+
+  const notesArray = notes.split('\n');
+
+  notesArray.forEach((part) => {
+    if (!part) {
+      return;
+    }
+
+    if (part.startsWith(HEADER_INDICATOR)) {
+      currentHeader = part
+        .replace(HEADER_INDICATOR, '')
+        .replace(
+          /([\u2700-\u27BF]|[\uE000-\uF8FF]|\uD83C[\uDC00-\uDFFF]|\uD83D[\uDC00-\uDFFF]|[\u2011-\u26FF]|\uD83E[\uDD10-\uDDFF])/g,
```

Comment:
Not clear to me what characters this is stripping away, can we add a short explaining comment?